core/src/main/scala/cats/TraverseFilter.scala

```diff
@@ -0,0 +1,116 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+ * `TraverseFilter`, also known as `Witherable` represents list-like structures
+ * that can essentially have a [[traverse]] and a [[filter]] applied as a single
+ * combined operation ([[traverseFilter]]).
+ *
+ * TODO ceedumbs more info
```

Comment:
By which I mean, mention in the comments the laws, I see they are encoded below in the tests.